---
title: "Calculate Rx Values - No Input"
output:
  html_document:
    df_print: paged
    theme: flatly
    number_sections: yes
    toc: yes
    toc_float: yes
    code_folding: hide
params:
  path_rx: '/lustre06/project/6004736/alvann/from_narval/20240304_EZHIP_OS_Revisions/ChIP/out/02A_genpipes_chipseq_mm10/metrics/SampleMetrics.tsv'
  path_sample: '/lustre06/project/6004736/alvann/from_narval/20240304_EZHIP_OS_Revisions/ChIP/out/01A_genpipes_chipseq/metrics/SampleMetrics.tsv'
  path_out: '/lustre06/project/6004736/alvann/from_narval/20240304_EZHIP_OS_Revisions/ChIP/out/02B_rx_values/without_input/'
  prefix: 'batch1.v3'
---

https://bio-protocol.org/en/bpdetail?id=2981&type=0

```{r}
library(tidyverse)
```

```{r}
notebook_theme = theme(aspect.ratio = 1,
                       text = element_text(size = 10), 
                       panel.border = element_rect(colour = "black", fill=NA, size=0.5),
                       panel.grid.major = element_blank(), 
                       panel.grid.minor = element_blank(),
                       panel.background = element_blank(),
                       axis.line = element_blank(), 
                       legend.position = "none",
                       axis.title = element_blank())
```

```{r}
print(params$path_genome)
print(params$path_rx) 
print(params$path_out) 
print(params$prefix)
```

```{r}
rx = data.table::fread(params$path_rx)
sample = data.table::fread(params$path_sample)
```

# Stats

```{r}
sample
```

```{r}
rx
```

# Calculate Rx

```{r}
sample = sample %>%
  mutate(Sample_Reads = `Final Aligned Reads # without Duplicates`) %>%
  select(Sample, `Mark Name`, Sample_Reads) %>%
  mutate(Sample_Reads_M = Sample_Reads/1e+6)
```


```{r}
rx = rx %>%
  mutate(Rx_Reads = `Final Aligned Reads # without Duplicates`) %>%
  select(Sample, `Mark Name`, Rx_Reads) %>%
  mutate(Rx_Reads_M = Rx_Reads/1e+6)
```

```{r}
norm = left_join(rx, sample, by = c('Sample', 'Mark Name')) %>%
  mutate(perc_sample = Sample_Reads/(Sample_Reads+Rx_Reads),
         rx = 1/Rx_Reads_M,
         hist_mod_level = rx*Rx_Reads)

norm  
```


```{r}
norm %>% 
  ggplot(aes(Sample, hist_mod_level)) +
    geom_bar(stat = 'identity', fill = '#558C8C') +
    facet_wrap(.~`Mark Name`) +
    theme(strip.background = element_blank(),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
    notebook_theme +
    scale_y_continuous(limits = c(0, max(norm$hist_mod_level) + 0.2), expand = c(0, 0)) 
```

```{r}
norm %>% 
  ggplot(aes(Sample, rx)) +
    geom_bar(stat = 'identity', fill = '#558C8C') +
    facet_wrap(.~`Mark Name`) +
    theme(strip.background = element_blank(),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
    notebook_theme +
    scale_y_continuous(limits = c(0, max(norm$rx) + 0.2), expand = c(0, 0)) 
```

```{r}
data.table::fwrite(norm, file = paste0(params$path_out, 'rx_norm.', params$prefix, '.txt'))
```
